### ISSUE TITLE
redundant tk.Menu get created for each item in bindings

### ISSUE THREAD

**Issue by pracio** [ragardner/tksheet#109]
first of all, thank you for this amazing library. i am currently using it for one of our project. it working beautifully.
second, this is my first time rasing issue to someone's repo, please forgive me if i did anything inappropriate.
ok, let's get back to the bug i found.
anytime when _tk_sheet_main_table.enable_bindings_internal was called, it will call self.create_rc_menus, thus, four tk.Menu will be created. so, if we're enabling bindings with self.enable_bindings, for each item in the bindings, it's create four tk.Menu, but only one is actually used.
i know the program runs just fine, at least it looks fine, the menu would popup. but it'll have back fire when the Sheet widget is being destroyed. each menu take 0.04s to destroy, as shown in the following logs
```
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu destroying time 0.04464316368103027
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu2 destroying time 0.04265618324279785
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu3 destroying time 0.0431516170501709
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu4 destroying time 0.044144630432128906
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu5 destroying time 0.0461275577545166
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu6 destroying time 0.044640541076660156
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu7 destroying time 0.044145822525024414
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu8 destroying time 0.0461273193359375
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu9 destroying time 0.0451357364654541
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu10 destroying time 0.043152809143066406
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu11 destroying time 0.04514336585998535
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu12 destroying time 0.0456235408782959
row index, .!toplevel.!toplevel.!tabularview.!sheet.!rowindex.!menu13 destroying time 0.048609018325805664

column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu destroying time 0.04315900802612305
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu2 destroying time 0.0436406135559082
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu3 destroying time 0.043648481369018555
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu4 destroying time 0.04564309120178223
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu5 destroying time 0.044145822525024414
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu6 destroying time 0.045151472091674805
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu7 destroying time 0.04364323616027832
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu8 destroying time 0.044133663177490234
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu9 destroying time 0.043648481369018555
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu10 destroying time 0.045135498046875
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu11 destroying time 0.04367566108703613
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu12 destroying time 0.04362010955810547
column headers, .!toplevel.!toplevel.!tabularview.!sheet.!columnheaders.!menu13 destroying time 0.04414796829223633
```
as you can see, there're 13 menus for both column header and row index, and my code for enable bindings as following
``` python
        # first menu item will be created when calling Sheet.__init__
        self.sheet.disable_bindings("all") # creates menu No.2
        self.sheet.enable_bindings((
            "single_select", # creats menu No.3
            "column_select", # creats menu No.4
            "row_select", # creats menu No.5
            "column_width_resize", # creats menu No.6
            "double_click_column_resize", # creats menu No.7
            "row_width_resize", # creats menu No.8
            "column_height_resize", # creats menu No.9
            "arrowkeys", # creats menu No.10
            "row_height_resize", # creats menu No.11
            "double_click_row_resize", # creats menu No.12
            "edit_cell")) # creats menu No.13
```
so my suggestion would be when creating the menus, we check if it's already exist. if so, we don't re-create them.
OK, that's all. thank you for reading, have a good day!

**Comment by ragardner** [comment]
Interesting, thanks for your great report and suggestions!

I figured old menus would be garbage collected when the variable was reassigned in `self.rc_popup_menu = tk.Menu(self, tearoff = 0, background = self.popup_menu_bg)` but I guess this is not the case?

It seems as though your pull request deals with everything correctly, I'll test it a bit and get back to you

**Comment by pracio** [comment]
> 

in the world of python, yes, self.rc_popup_menu would be garbage collected, however tkinter is just a thin wrapper that wraps around tk/tcl they have their own ways tracking references, so, in guess in the world of tk/tcl self.rc_popup_menu is still a child of maintable.
i tried calling self.rc_popup_menu.destroy() right before we reassign it to a new instance and it works. it does cost less time when the parent widget get destroyed. however, due to the way how enable_bindings/disable_bindings and create_rc_menu is implemented, it certainly takes longer to set up the menu, since the destruction of menu is actully the performance bottle neck in my case. and that's also the reason why in my pull request #110  i choose 're-add' commands to the menu instead of destroying/creating a new one.

**Comment by ragardner** [comment]
@pracio Thanks very much for the info! I should've figured this is the case... 

I have released your suggested changes in `5.3.1` (including your `disable_bindings` menu clear up suggestion, thanks!)

**Comment by pracio** [comment]
> @pracio Thanks very much for the info! I should've figured this is the case...
> 
> I have released your suggested changes in `5.3.1` (including your `disable_bindings` menu clear up suggestion, thanks!)

awesome! i tested the version you pushed to pip, it works as expected! :)
